src/pages/ServiceDetails/ServiceInfo/ServiceInfoDestinationPolicies.tsx

```diff
@@ -0,0 +1,122 @@
+              </div>
+            );
+          }
+          let loadBalancing;
+          if (dPolicy.loadbalancing) {
+            loadBalancing = (
+              <div>
+                <strong>LoadBalancing</strong>
+                {': '}
+                {dPolicy.loadbalancing.name}
+              </div>
+            );
+          }
+          let circuitBreaker;
+          if (dPolicy.circuitBreaker) {
+            circuitBreaker = (
+              <div>
+                <strong>CircuitBreaker</strong>
+                {': simpleCb'}
+                <ul style={{ listStyleType: 'none' }}>
+                  {!dPolicy.circuitBreaker.simpleCb.maxConnections ? null : (
+                    <li>maxConnections: {dPolicy.circuitBreaker.simpleCb.maxConnections}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpMaxPendingRequests ? null : (
+                    <li>httpMaxPendingRequests: {dPolicy.circuitBreaker.simpleCb.httpMaxPendingRequests}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpMaxRequests ? null : (
+                    <li>httpMaxRequests: {dPolicy.circuitBreaker.simpleCb.httpMaxRequests}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.sleepWindow ? null : (
+                    <li>sleepWindow: {dPolicy.circuitBreaker.simpleCb.sleepWindow}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpConsecutiveErrors ? null : (
+                    <li>httpConsecutiveErrors: {dPolicy.circuitBreaker.simpleCb.httpConsecutiveErrors}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpDetectionInterval ? null : (
+                    <li>httpDetectionInterval: {dPolicy.circuitBreaker.simpleCb.httpDetectionInterval}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpMaxRequestsPerConnection ? null : (
+                    <li>
+                      httpMaxRequestsPerConnection: {dPolicy.circuitBreaker.simpleCb.httpMaxRequestsPerConnection}
+                    </li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpMaxEjectionPercent ? null : (
+                    <li>httpMaxEjectionPercent: {dPolicy.circuitBreaker.simpleCb.httpMaxEjectionPercent}</li>
+                  )}
+                  {!dPolicy.circuitBreaker.simpleCb.httpMaxRetries ? null : (
+                    <li>httpMaxRetries: {dPolicy.circuitBreaker.simpleCb.httpMaxRetries}</li>
+                  )}
+                </ul>
+              </div>
+            );
+          }
+          return (
+            <div key={'rule' + i}>
+              <div>
+                <strong>Name</strong>
+                {': '}
+                {dPolicy.name}
+              </div>
+              {dPolicySource}
+              {destinationLabels}
+              {loadBalancing}
+              {circuitBreaker}
```

Comment:
Would it be useful to split this render in different components? Having one per each last 2 lines above (LoadBalancing, CircuitBreaker). I think this way we let those 2 parts evolve independently of this component and making modifications easier.
I am not very familiar with React already, so maybe I am over modularizing this.

Reply:
Probably, but I wouldn't introduce that refactor on this PR as it would depend on other factor.
Also, LoadBalancing and CircuitBreaker have not enough entity yet to split them in different components IMO.

Reply:
I think in a not far future we should try to make those conceptual entities containers, for a better maintainability.